Core/src/org/sleuthkit/autopsy/filequery/FileSearch.java

```diff
@@ -746,6 +836,516 @@ public int hashCode() {
         }    
     }    
     
+    /**
+     * Attribute for grouping/sorting by hash set lists
+     */
+    static class HashHitsAttribute extends AttributeType {
+        
+        @Override
+        GroupKey getGroupKey(ResultFile file) {
+            return new HashHitsGroupKey(file);
+        }
+        
+        @Override
+        void addAttributeToResultFiles(List<ResultFile> files, SleuthkitCase caseDb, 
+                EamDb centralRepoDb) throws FileSearchException {
+            
+            // Get pairs of (object ID, keyword list name) for all files in the list of files that have
+            // hash set hits.
+            String selectQuery = createSetNameQuery(files, caseDb, BlackboardArtifact.ARTIFACT_TYPE.TSK_HASHSET_HIT.getTypeID(),
+                    BlackboardAttribute.ATTRIBUTE_TYPE.TSK_SET_NAME.getTypeID());
+            
+            SetHashSetNamesCallback callback = new SetHashSetNamesCallback(files);
+            try {
+                caseDb.getCaseDbAccessManager().select(selectQuery, callback);
+            } catch (TskCoreException ex) {
+                throw new FileSearchException("Error looking up hash set attributes", ex); // NON-NLS
+            }
+        }
+        
+        /**
+         * Callback to process the results of the CaseDbAccessManager select query. Will add
+         * the keyword list names to the list of ResultFile objects.
+         */
+        private static class SetHashSetNamesCallback implements CaseDbAccessManager.CaseDbAccessQueryCallback {
```

Comment:
The overloading of "set" in this class name makes this name a little hard to parse. Perhaps we don't need the first "Set" in this and the other callback names.